prediction/src/algorithms/segment/trained_model.py

```diff
@@ -38,17 +41,26 @@ def predict(dicom_path, centroids):
             {'binary_mask_path': str,
              'volumes': list[float]}
     """
-    load_ct(dicom_path)
-    segment_path = os.path.join(os.path.dirname(__file__),
-                                'assets', 'test_mask.npy')
+    voxel_data, meta = load_ct(dicom_path)
+    model = load_model(get_best_model_path(), custom_objects={'dice_coef_loss': dice_coef_loss, 'dice_coef': dice_coef})
+    x, y, z, channels = get_data_shape()
+    input_data = np.ndarray((1, x, y, z, channels))  # batch, x, y, z, channels
+    # Crop the input data to the required data shape and pad with zeros
+    padded_data = np.zeros_like(input_data)
+    min_x, min_y, min_z = min(x, voxel_data.shape[0]), min(y, voxel_data.shape[1]), min(z, voxel_data.shape[2])
+    padded_data[0, :min_x, :min_y, :min_z, 0] = voxel_data[:min_x, :min_y, :min_z]
+    input_data = padded_data
+
+    output_data = model.predict(input_data)
+    segment_path = os.path.join(os.path.dirname(__file__), 'assets', "lung-mask.npy")
```

Comment:
Would prefer more constants than the use of underscoreunderscorefileunderscoreunderscore :)

Reply:
fixed in 8416d58